### PR TITLE
Added prototype of client REST application with SpringBoot and RabbitMQ

### PR DESCRIPTION
**Pull request by JaroslawSlaby**
Prototype of web client app

### CODE REVIEW

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/configuration/RabbitMQConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  @Value("${queue.gameConfig}")
+  String gameConfigQueueName;
+
+  @Value("${queue.playerConfig}")
+  String playerConfigQueueName;
+
+  @Bean
+  public Queue getGameConfigQueue() {
+    return new Queue(gameConfigQueueName, false, false, false);
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/configuration/RabbitMQConfiguration.java]: extract variables

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/MainController.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package pl.jakubowskiprzemyslaw.REST.controllers;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import pl.jakubowskiprzemyslaw.REST.models.Game;
+
+@Controller
+public class MainController {
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/MainController.java]: StartController

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/MainController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  private int numberOfConnections = 0;
+
+  @GetMapping(value = "/start", produces = "text/html")
+  public String getMainPage() {
+    //TODO: 4.07.2018 - This value should be taken from REST API form micro service GameConfigurationRegisterer
+    numberOfConnections++;
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/MainController.java]: ammountOfConnectedPlayers

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/configuration/RabbitMQConfiguration.java (hunk 1 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/configuration/RabbitMQConfiguration.java]: done

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/MainController.java (hunk 2 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/MainController.java]: done

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/MainController.java (hunk 3 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/MainController.java]: done

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  private String gameConfigQueueName;
+
+  @Autowired
+  public GameConfigController(QueueService queueService, SessionService sessionService) {
+    super(queueService, sessionService);
+  }
+
+  @GetMapping(value = "/gameconfig", produces = "text/html")
+  public String getGameConfig(Model model) {
+    model.addAttribute("configuration", new Game());
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java]: new GameConfiguration()

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/FleetPlacementController.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+public class FleetPlacementController extends RESTController {
+
+  @Autowired
+  public FleetPlacementController(QueueService queueService, SessionService sessionService) {
+    super(queueService, sessionService);
+  }
+
+  @GetMapping(value = "/fleetplacement")
+  public String getFleetPlacement(Model model, HttpServletRequest request) {
+    if(!isObjectInSession(request, "Player"))
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/FleetPlacementController.java]: add moustaches to if

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    return "gameconfig";
+  }
+
+  @PostMapping(value = "/gameconfig", produces = "text/html")
+  public String sendGameConfig(@Valid @ModelAttribute("configuration") Game configuration, HttpServletRequest request) {
+    addObjectToSession(request, configuration);
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java]: you're adding object to session, but you're sending request from which session is extracted later.
Session should be extracted here and then passed to method, or method should be renamed (first option preffered)

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    sendMessageToQueue(gameConfigQueueName, configuration);
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java]: it's rather sendObjectToQueue

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/NoPlacesController.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package pl.jakubowskiprzemyslaw.REST.controllers;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class NoPlacesController {
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/NoPlacesController.java]: GameFullController

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/NoPlacesController.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+  @GetMapping
+  public String noFreePlaces() {
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/NoPlacesController.java]: gameFull()

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/PlayerConfigController.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import pl.jakubowskiprzemyslaw.REST.models.Player;
+import pl.jakubowskiprzemyslaw.REST.services.QueueService;
+import pl.jakubowskiprzemyslaw.REST.services.SessionService;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Controller
+public class PlayerConfigController extends RESTController {
+
+  @Value("${queue.playerConfig}")
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/PlayerConfigController.java]: Those properties are in fact queueName.playerConfig and so on, not queues itself. We should think about better names for those

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/RESTController.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  private final QueueService queueService;
+  private final SessionService sessionService;
+
+  RESTController(QueueService queueService, SessionService sessionService) {
+    this.queueService = queueService;
+    this.sessionService = sessionService;
+  }
+
+  void sendMessageToQueue(String queueName, Object message) {
+    queueService.sendMessageToQueue(queueName, message);
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/RESTController.java]: Object != message

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java (hunk 6 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java]: second version chosen and done

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java (hunk 7 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/GameConfigController.java]: done

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/services/SessionService.java [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+@Service
+public class SessionService {
+
+  public void addObjectToSession(HttpServletRequest request, Object object) {
+    HttpSession session = request.getSession();
+    session.setAttribute(object.getClass().getSimpleName(), object);
+  }
+
+  public boolean isObjectInSession(HttpServletRequest request, String objectName) {
```
Comment (boczkas) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/services/SessionService.java]: you're sending request, but name of session suggests that it should accept request

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/NoPlacesController.java (hunk 8 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/NoPlacesController.java]: done

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/NoPlacesController.java (hunk 9 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/NoPlacesController.java]: done

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/PlayerConfigController.java (hunk 10 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/PlayerConfigController.java]: changed to queueName.playerConfig and other properties for queues look the same

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/RESTController.java (hunk 11 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/controllers/RESTController.java]: done

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/models/Player.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
+package pl.jakubowskiprzemyslaw.REST.models;
+
+import java.io.Serializable;
+
+public class Player implements Serializable { //TODO: 4.07.2018 - add more game configuration parameters according to requirements
```
[the remark this answers is not in the document]
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/models/Player.java]: generated and added to both classes

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/services/SessionService.java (hunk 12 above)
Comment (JaroslawSlaby) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/services/SessionService.java]: changed

**Pull request closed without merging**